### PR TITLE
Update .luacheckrc

### PR DESCRIPTION
**Pull request by cooljeanius**
This PR implements the suggestion from @ProditorMagnus in #90

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -4,9 +4,9 @@ codes=true
 -- skip showing files with no issues
 quiet=1
 -- skip showing undefined variable usage
--- there are thousands of warnings here because luacheck is unaware of Wesnoth's
+-- there can be warnings because luacheck is unaware of Wesnoth's
 -- lua environment and has no way to check which have been loaded
-global=false
+globals={"wesnoth","wml","filesystem","start_x","start_y","delta_x","delta_y","index","helper"}
```
Comment (ProditorMagnus) [inline .luacheckrc]: this list contains wrong entries. start_x is not provided by wesnoth for example

Comment (cooljeanius) [reply .luacheckrc]: I had to add all of these to get the check to pass

Comment (ProditorMagnus) [reply .luacheckrc]: https://luacheck.readthedocs.io/en/stable/warnings.html allow_defined  sounds like it could solve code style related 111 and 113

Comment (ProditorMagnus) [reply .luacheckrc]: incorrectly passing check is worse than failing check

Comment (ProditorMagnus) [reply .luacheckrc]: based on emmylua wesnoth provides these globals   "Lua.diagnostics.globals": [
        "wesnoth",
        "wml",
        "gui",
        "filesystem",
        "unit_test",
        "stringx",
        "mathx",
        "ai"
    ]

Comment (ProditorMagnus) [reply .luacheckrc]: code incorrectly uses helper, so check has to fail, otherwise check is useless

Comment (cooljeanius) [reply .luacheckrc]: I don't know how to fix the incorrect use of helper, though...

Comment (ProditorMagnus) [reply .luacheckrc]: thats ok, not topic for this pr

**Pull request merged**